### PR TITLE
Introduce the version constraint of scikit-learn

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Scikit-learn 1.0.0 is out. Unfortunately, scikit-optimize does not work with the latest scikit-learn.
CI error example: https://github.com/nzw0301/optuna/runs/3701260684?check_suite_focus=true
See also the reported issue to scikit-optimize repo: https://github.com/scikit-optimize/scikit-optimize/issues/1059

## Description of the changes

Introduce the version constraint of scikit-learn and TODO lines to the config and setup files that use scikit-optimize.

### CODE REVIEW

File: .github/workflows/mac-tests.yml [hunk 1]
```diff
@@ -46,6 +46,11 @@ jobs:
         optuna --version
         pip install --progress-bar off .[tests]
         pip install --progress-bar off .[optional]
+
+        # TODO(nzw0301) remove the constraint of sklearn `1.0.0`
+        # after skopt supports sklearn >= 1.0.0
+        pip install "scikit-learn<1.0.0"
```
Comment (hvy) [inline .github/workflows/mac-tests.yml:52]: Can or should we not update the `optional` extras requirements in `setup.py` instead?

**Review by hvy — COMMENTED** [review]
Thanks for the quick fix. I left a single question if you could take a look.

File: .github/workflows/mac-tests.yml (hunk 1 above)
Comment (nzw0301) [reply .github/workflows/mac-tests.yml]: Definitely. I've updated it. Thanks!

**Review by hvy — APPROVED** [review]
Thanks for the prompt response, LGTM!

**Review by not522 — CHANGES_REQUESTED** [review]
Could you add constraints in document section's scikit-learn?

**Review by not522 — APPROVED** [review]
Thank you. LGTM!

**Pull request merged**